### PR TITLE
{bugfix} Prevent field conflicts in Join by Nearest

### LINKED ISSUE #44904: [Bug] Join by nearest, overflow in case of field conflicts
### What is the bug or the crash?

When using Join by Nearest with a dataset that already has one or more field from the alg, the following fields are offest.

Basically if the field 'n' is present, the values from 'n' will be in 'distance' and 'distance' in 'feature_x', etc.

### Steps to reproduce the issue

Perform a Join by Nearest  on two layers.

Use the resulting dataset to perform another Join by Nearest on the result from the previous run.

### Versions

Tested on a month old master.

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

Unsure if the content should be entirely overridden, the incoming content discarded or new fields created.

### PR DESCRIPTION
**Pull request by roya0045**
## Description

Proposal to handle conflicting fields in the Join by Nearest algorithm.

Fixes #44904 

I'm unsure about the test, I would either need to create a new test, or edit the test data to add a new field.

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmjoinbynearest.cpp [hunk 1]
```diff
@@ -169,12 +170,21 @@ QVariantMap QgsJoinByNearestAlgorithm::processAlgorithm( const QVariantMap &para
   const QgsAttributeList fields2Fetch = fields2Indices;
 
   QgsFields outFields = QgsProcessingUtils::combineFields( input->fields(), outFields2 );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmjoinbynearest.cpp:172]: Better to just use combineFields again here, i.e.:

```
  QgsFields additionalFields;
  additionalFields.append( QgsField( QStringLiteral( "n" ), QVariant::Int ) );
  additionalFields.append( QgsField( QStringLiteral( "distance" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "feature_x" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "feature_y" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "nearest_x" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "nearest_y" ), QVariant::Double ) ); 
  outFields = QgsProcessingUtils::combineFields( outFields, additionalFields );
```

Comment (roya0045) [reply src/analysis/processing/qgsalgorithmjoinbynearest.cpp:172]: I overlooked the fact that `combineFields` already has the conflict handling logic. Thanks!

**Comment by roya0045** [comment]
Looks like the processing tests broke in general, though seems unrelated to this PR as other are affected.

**Comment by nyalldawson** [comment]
> Looks like the processing tests broke in general, though seems unrelated to this PR as other are affected.

There's a lengthy test causing timeouts -- it's fixed in https://github.com/qgis/QGIS/pull/46579, but the whole CI infrastructure is down now anyway. I suggest we all just knock off for Christmas early :wink:

**Comment by roya0045** [comment]
> > Looks like the processing tests broke in general, though seems unrelated to this PR as other are affected.
> 
> There's a lengthy test causing timeouts -- it's fixed in #46579, but the whole CI infrastructure is down now anyway. I suggest we all just knock off for Christmas early 😉

This'll give me more time to fix my build and work on more complex bugs if I have the time.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**